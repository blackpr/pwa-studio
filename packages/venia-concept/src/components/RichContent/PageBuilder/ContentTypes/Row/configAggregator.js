import { getBackgroundImages, getVerticalAlignment } from '../../utils';

export default node => {
    console.log(node);
    return {
        minHeight: node.childNodes[0].style.minHeight ? node.childNodes[0].style.minHeight : null,
        ...getVerticalAlignment(node.childNodes[0]),
        backgroundColor: node.childNodes[0].style.backgroundColor,
        ...getBackgroundImages(node.childNodes[0]),
        enableParallax: node.childNodes[0].getAttribute('data-enable-parallax') === "1",
        parallaxSpeed: parseFloat(
            node.childNodes[0].getAttribute('data-parallax-speed')
        ),
        cssClasses: node.childNodes[0].getAttribute('class'),
        border: node.childNodes[0].style.borderStyle,
        borderColor: node.childNodes[0].style.borderColor,
        borderWidth: node.childNodes[0].style.borderWidth,
        borderRadius: node.childNodes[0].style.borderRadius,
        marginTop: node.childNodes[0].style.marginTop,
        marginRight: node.childNodes[0].style.marginRight,
        marginBottom: node.childNodes[0].style.marginBottom,
        marginLeft: node.childNodes[0].style.marginLeft,
        paddingTop: node.childNodes[0].style.paddingTop,
        paddingRight: node.childNodes[0].style.paddingRight,
        paddingBottom: node.childNodes[0].style.paddingBottom,
        paddingLeft: node.childNodes[0].style.paddingLeft,
    };
};
