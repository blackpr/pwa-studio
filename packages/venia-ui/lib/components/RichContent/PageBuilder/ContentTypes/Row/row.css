.contained {
    box-sizing: border-box;
    margin-left: auto !important;
    margin-right: auto !important;
    max-width: 1280px;
    width: 100%;
}
