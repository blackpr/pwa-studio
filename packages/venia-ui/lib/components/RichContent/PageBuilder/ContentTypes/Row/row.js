import React from 'react';
import defaultClasses from './row.css';
import { verticalAlignmentToFlex } from '../../utils';
import { Parallax } from 'react-parallax';
import { mergeClasses } from '../../../../../classify';

const Row = ({
    classes,
    appearance,
    verticalAlignment,
    minHeight,
    backgroundColor,
    desktopImage,
    mobileImage,
    backgroundSize,
    backgroundPosition,
    backgroundAttachment,
    backgroundRepeat,
    enableParallax,
    parallaxSpeed,
    textAlign,
    border,
    borderColor,
    borderWidth,
    borderRadius,
    marginTop,
    marginRight,
    marginBottom,
    marginLeft,
    paddingTop,
    paddingRight,
    paddingBottom,
    paddingLeft,
    cssClasses,
    children
}) => {
    classes = mergeClasses(defaultClasses, classes);
    // Set the default appearance if none is supplied to contained
    appearance = appearance ? appearance : 'contained';
    let image = desktopImage;
    if (mobileImage && window.matchMedia('(max-width: 768px)').matches) {
        image = mobileImage;
    }
    const dynamicStyles = {
        minHeight: minHeight,
        backgroundColor: backgroundColor,
        textAlign,
        border,
        borderColor,
        borderWidth,
        borderRadius,
        marginTop,
        marginRight,
        marginBottom,
        marginLeft,
        paddingTop,
        paddingRight,
        paddingBottom,
        paddingLeft
    };

    if (!enableParallax) {
        dynamicStyles.backgroundImage = image ? `url(${image})` : null;
        dynamicStyles.backgroundSize = backgroundSize;
        dynamicStyles.backgroundPosition = backgroundPosition;
        dynamicStyles.backgroundAttachment = backgroundAttachment;
        dynamicStyles.backgroundRepeat = backgroundRepeat
            ? 'repeat'
            : 'no-repeat';
    }

    if (verticalAlignment) {
        dynamicStyles.display = 'flex';
        dynamicStyles.justifyContent = verticalAlignmentToFlex(
            verticalAlignment
        );
        dynamicStyles.flexDirection = 'column';
    }

    // Full width and contained appearance
    if (appearance === 'contained') {
        cssClasses.push(classes.contained);
    }
    if (appearance === 'full-width') {
        children = <div className={classes.contained}>{children}</div>;
    }

    if (enableParallax) {
        return (
            <Parallax
                strength={parallaxSpeed * 200}
                bgImage={image}
                style={dynamicStyles}
                className={cssClasses.join(' ')}
            >
                {children}
            </Parallax>
        );
    }

    return (
        <div style={dynamicStyles} className={cssClasses.join(' ')}>
            {children}
        </div>
    );
};

export default Row;
