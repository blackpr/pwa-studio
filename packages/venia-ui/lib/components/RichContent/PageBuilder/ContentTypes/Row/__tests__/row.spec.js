import React from 'react';
import testRenderer from 'react-test-renderer';
import Row from '../row';

jest.mock('../../../../../../classify');

test('renders a Row component', () => {
    const component = testRenderer.create(<Row />);

    expect(component.toJSON()).toMatchSnapshot();
});

test('renders a Row component with all props configured', () => {
    const rowProps = {
        appearance: 'full-width',
        verticalAlignment: 'middle',
        minHeight: '200px',
        backgroundColor: 'red',
        desktopImage: 'desktop.jpg',
        mobileImage: 'mobile.jpg',
        backgroundSize: 'contain',
        backgroundPosition: 'center center',
        backgroundAttachment: 'fixed',
        backgroundRepeat: true,
        enableParallax: true,
        parallaxSpeed: 0.5,
        textAlign: 'right',
        border: 'solid',
        borderColor: 'red',
        borderWidth: '10px',
        borderRadius: '15px',
        marginTop: '10px',
        marginRight: '10px',
        marginBottom: '10px',
        marginLeft: '10px',
        paddingTop: '10px',
        paddingRight: '10px',
        paddingBottom: '10px',
        paddingLeft: '10px',
        cssClasses: ['test-class']
    };
    const component = testRenderer.create(<Row {...rowProps} />);

    expect(component.toJSON()).toMatchSnapshot();
});
