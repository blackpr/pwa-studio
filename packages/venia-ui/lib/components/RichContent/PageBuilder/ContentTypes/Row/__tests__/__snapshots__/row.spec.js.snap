// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Row component 1`] = `
<div
  className="contained"
  style={
    Object {
      "backgroundAttachment": undefined,
      "backgroundColor": undefined,
      "backgroundImage": null,
      "backgroundPosition": undefined,
      "backgroundRepeat": "no-repeat",
      "backgroundSize": undefined,
      "border": undefined,
      "borderColor": undefined,
      "borderRadius": undefined,
      "borderWidth": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "minHeight": undefined,
      "paddingBottom": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "textAlign": undefined,
    }
  }
/>
`;

exports[`renders a Row component with all props configured 1`] = `
<div
  className="react-parallax test-class"
  style={
    Object {
      "backgroundColor": "red",
      "border": "solid",
      "borderColor": "red",
      "borderRadius": "15px",
      "borderWidth": "10px",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "marginBottom": "10px",
      "marginLeft": "10px",
      "marginRight": "10px",
      "marginTop": "10px",
      "minHeight": "200px",
      "overflow": "hidden",
      "paddingBottom": "10px",
      "paddingLeft": "10px",
      "paddingRight": "10px",
      "paddingTop": "10px",
      "position": "relative",
      "textAlign": "right",
    }
  }
>
  <img
    alt=""
    className="react-parallax-bgimage"
    src="desktop.jpg"
    style={
      Object {
        "MozBackfaceVisibility": "hidden",
        "MsBackfaceVisibility": "hidden",
        "WebkitBackfaceVisibility": "hidden",
        "WebkitTransform": "translate3d(-50%, 0, 0)",
        "WebkitTransformStyle": "preserve-3d",
        "left": "50%",
        "position": "absolute",
        "transform": "translate3d(-50%, 0, 0)",
      }
    }
  />
  <div
    className="react-parallax-content"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="contained"
    />
  </div>
</div>
`;
